crates/model/src/signature.rs

```diff
@@ -327,31 +327,37 @@ impl EcdsaSignature {
         struct_hash: &[u8; 32],
     ) -> Result<Recovered> {
         let message = hashed_signing_message(signing_scheme, domain_separator, struct_hash);
-        let recovery = Recovery::new(message, self.v as u64, self.r.0.into(), self.s.0.into());
+        let recovery = Recovery::new(
+            message.0,
+            self.v as u64,
+            H256(B256::from(self.r).0),
+            H256(B256::from(self.s).0),
+        );
         let (signature, recovery_id) = recovery
             .as_signature()
             .context("unexpectedly invalid signature")?;
-        let signer = Address::new(signing::recover(&message, &signature, recovery_id)?.0);
+        let signer = Address::new(signing::recover(&message.0, &signature, recovery_id)?.0);
 
-        Ok(Recovered {
-            message: message.into(),
-            signer,
-        })
+        Ok(Recovered { message, signer })
     }
 
     pub fn sign(
         signing_scheme: EcdsaSigningScheme,
         domain_separator: &DomainSeparator,
         struct_hash: &[u8; 32],
-        key: SecretKeyRef,
+        key: &PrivateKeySigner,
     ) -> Self {
         let message = hashed_signing_message(signing_scheme, domain_separator, struct_hash);
         // Unwrap because the only error is for invalid messages which we don't create.
-        let signature = key.sign(&message, None).unwrap();
+        let signature = key.sign_hash_sync(&message).unwrap();
```

Comment:
This could be replaced with `Self::from_bytes(&signature.as_bytes())` as this already handles the magic number inside `as_bytes()`.